sale_substate/models/sale_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class BaseSubstateType(models.Model):
+    _inherit = "base.substate.type"
+
+    model = fields.Selection(
+        selection_add=[("sale.order", "Sale order")], ondelete={"sale.order": "cascade"}
+    )
+
+
+class SaleOrder(models.Model):
+    _inherit = ["sale.order", "base.substate.mixin"]
+    _name = "sale.order"
+
+    @api.constrains("substate_id", "state")
+    def check_substate_id_value(self):
```

Comment:
@Wodran14 IMHO this is covered by : https://github.com/OCA/server-ux/pull/855/files#diff-84ca126b7a08243bd6268af8f3c6e2ffe5ffa1e2d82b9cdb867840cd14a0b2f4R14